File: src/WorkflowShell/git/GitFixIgnore.rb
require_relative '../models/Command'

class GitAddDot < Command
  @command_string
  @command_description
  @command_usage

  def initialize
    @command_string = 'gitfixignore'
    @command_description = 'Attempts to fix gitignore not working properly.'
    @command_usage = @command_string
  end

  def run_command(arguments)
    # Ask the Command class to parse the basic options like help and verbosity
    parser_components = parse_options(arguments, true)

    run_shell_command("git commit -a -m \"Before Fixing gitignore\"", parser_components.basic_options.verbose)
    run_shell_command("git rm -r --cached .", parser_components.basic_options.verbose)
    run_shell_command("git add .", parser_components.basic_options.verbose)
    run_shell_command("git commit -a -m \"Fixed gitignore\"", parser_components.basic_options.verbose)
  end
end

File: src/WorkflowShell/git/GitFixIgnore.ts
import Command from "../models/Command";

export default class GitFixIgnore extends Command {

  static getString(): string {
    return "gfi";
  }

  static getDescription(): string {
    return "Attempts to fix gitignore not working properly.";
  }

  getUsage(): string {
    return `${GitFixIgnore.getString()}`;
  }

  async run(args: any) {
    const ok: boolean = await super.run(args);
    if (!ok) {
      return false;
    }

    super.execute("git commit -a -m \"Before Fixing gitignore\"");
    super.execute("git rm -r --cached .");
    super.execute("git add .");
    super.execute("git commit -a -m \"Fixed gitignore\"");
    return true;
  }
}
